packages/vite/src/node/plugins/html.ts

```diff
@@ -588,7 +588,7 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
           }
         }
 
-        const shortEmitName = path.posix.relative(config.root, id)
+        const shortEmitName = path.relative(config.root, id)
```

Comment:
I think we should use the same construct as in https://github.com/vitejs/vite/blob/0646fe8661b2d0a46eaff6a752b3108daab82c46/packages/vite/src/node/plugins/html.ts#L234
And also add the `/` directly in this line

Reply:
When I add the `/`, my minimal reproduction fails with the same error message as before:

```
$ npm run build

> vite-project@0.0.0 build
> vite build

vite v2.8.6 building for production...
✓ 1 modules transformed.
Generated an empty chunk: "index"
rendering chunks (1)...[vite:build-html] The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths, received "/index.html". 
error during build:
Error: The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths, received "/index.html".
    at error (C:\Users\Johannes\Code\vite-build-path-windows-reproduction\node_modules\rollup\dist\shared\rollup.js:198:30)
    at FileEmitter.emitFile (C:\Users\Johannes\Code\vite-build-path-windows-reproduction\node_modules\rollup\dist\shared\rollup.js:15513:24)
    at Object.generateBundle (C:\Users\Johannes\Code\vite-build-path-windows-reproduction\node_modules\vite\dist\node\chunks\dep-9c153816.js:21677:14)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

Reply:
> The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths, received "/index.html".

This error is a little obsure, but what it means is that when using `this.emitFile`, `name` or `fileName` cannot start with a `/` (absolute path) or a `./` or `../` (relative path)